diff --git a/dwm.c b/dwm.c
index b8f577b..1f47c19 100644
--- a/dwm.c
+++ b/dwm.c
@@ -924,7 +924,7 @@ drawbar(Monitor *m)
 		stw = getsystraywidth();
 
 	/* draw status first so it can be overdrawn by tags later */
-	if (m == selmon) { /* status is only drawn on selected monitor */
+	if (m == selmon || 1) { /* status is only drawn on selected monitor */
 		drw_setscheme(drw, scheme[SchemeNorm]);
 		sw = TEXTW(stext) - lrpad / 2 + 2; /* 2px right padding */
 		drw_text(drw, m->ww - sw - stw, 0, sw, bh, lrpad / 2 - 2, stext, 0);
@@ -2395,10 +2395,13 @@ updatesizehints(Client *c)
 void
 updatestatus(void)
 {
+  Monitor* m;
+
   if (!gettextprop(root, XA_WM_NAME, stext, sizeof(stext))) {
 		strcpy(stext, "dwm-"VERSION);
   }
-	drawbar(selmon);
+  for(m = mons; m; m = m->next)
+    drawbar(m);
 	updatesystray();
 }
 
